### PR TITLE
Add position parameter to the PadIfNeeded transform

### PR DESCRIPTION
**Pull request by yisaienkov**
Problem described in this issue #927 

I add an option to put the image to the top left, top right, bottom left, bottom right, or center positions.

### CODE REVIEW

**Comment by yisaienkov** [comment]
@creafz I think your ci pipeline updated numpy version and all your code is not mypy correct now
The previous pipeline has this output in "Install PyTorch on Linux and Windows" step:
`Successfully installed numpy-1.20.3 pillow-8.2.0 torch-1.8.1+cpu torchvision-0.9.1+cpu typing-extensions-3.10.0.0`
But now:
`Successfully installed numpy-1.21.0 pillow-8.2.0 torch-1.8.1+cpu torchvision-0.9.1+cpu typing-extensions-3.10.0.0`

File: albumentations/augmentations/transforms.py [hunk 1]
```diff
@@ -107,6 +109,7 @@ def __init__(
         min_width: Optional[int] = 1024,
         pad_height_divisor: Optional[int] = None,
         pad_width_divisor: Optional[int] = None,
+        position: str = "center",
```
Comment (Dipet) [inline albumentations/augmentations/transforms.py:112]: Maybe would be better to use `Enum` for position type? You could describe it as a class parameter.

File: albumentations/augmentations/transforms.py [hunk 2]
```diff
@@ -214,6 +226,35 @@ def get_transform_init_args_names(self):
...
+            h_top += h_bottom
+            w_right += w_left
+            h_bottom = 0
+            w_left = 0
+
+        if self.position == "bottom_right":
+            h_top += h_bottom
+            w_left += w_right
+            h_bottom = 0
+            w_right = 0
```
Comment (Dipet) [inline albumentations/augmentations/transforms.py:254]: I think it better to use `elif` constraction there.

File: tests/test_augmentations.py [hunk 3]
```diff
@@ -714,6 +714,46 @@ def test_pad_if_needed(augmentation_cls: Type[PadIfNeeded], params: Dict, image_
...
+        true_result[: image_shape[0], -image_shape[1] :] = 0
+        assert (image_padded == true_result).all()
+
+    if params["position"] == "bottom_left":
+        true_result[-image_shape[0] :, : image_shape[1]] = 0
+        assert (image_padded == true_result).all()
+
+    if params["position"] == "bottom_right":
+        true_result[-image_shape[0] :, -image_shape[1] :] = 0
+        assert (image_padded == true_result).all()
```
Comment (Dipet) [inline tests/test_augmentations.py:754]: I think it better to use `elif` construction there.

File: albumentations/augmentations/transforms.py (hunk 1 above)
Comment (yisaienkov) [reply albumentations/augmentations/transforms.py]: @Dipet added, can you check, please?

File: albumentations/augmentations/transforms.py (hunk 2 above)
Comment (yisaienkov) [reply albumentations/augmentations/transforms.py]: @Dipet fixed, can you check, please?

File: tests/test_augmentations.py (hunk 3 above)
Comment (yisaienkov) [reply tests/test_augmentations.py]: @Dipet I fixed it, can you check, please?

**Review by creafz — APPROVED** [review]

**Review by ternaus — APPROVED** [review]

**Pull request merged**